salt/states/boto_vpc.py

```diff
@@ -1081,3 +1095,140 @@ def route_table_absent(name, region=None,
     ret['changes']['new'] = {'route_table': None}
     ret['comment'] = 'Route table {0} deleted.'.format(name)
     return ret
+
+
+def nat_gateway_present(name, subnet_name=None, subnet_id=None,
+                        region=None, key=None, keyid=None, profile=None):
+    '''
+    Ensure a nat gateway exists within the specified subnet
```

Comment:
This docstring could also use a boto3 dependency note as well as a `.. versionadded:: Carbon` directive.